docs/BSD.md

```diff
@@ -0,0 +1,84 @@
+# BSD building/testing instructions
+
+BSD is not officially supported.
+These are some basic instructions for getting CFEngine Community to build on FreeBSD.
+Tested on FreeBSD 11 VM in AWS EC2.
+
+## Dependencies
+
+```
+$ su root -c 'pkg update -f'
+$ su root -c 'pkg install -y gdb gcc gmake lmdb autoconf automake libtool git python3'
+```
+
+## Download
+
+```
+$ git clone --recursive https://github.com/cfengine/core.git
+$ git clone --recursive https://github.com/cfengine/masterfiles.git
+```
+
+## Install masterfiles
+
+```
+$ (cd masterfiles && ./autogen.sh && su root -c "gmake install")
+```
+
+## Check out PR (Optional)
+
+To check out PR `#3589` to branch `bsd-branch`, do:
+
+```
+$ cd core
+$ git fetch origin pull/3589/head:bsd-branch
+$ git checkout bsd-branch
+```
+
+## Build
+
+Within `core` directory:
+
+```
+$ ./autogen.sh --enable-debug -C --with-lmdb=/usr/local/ --with-pcre=/usr/local/
+$ gmake -j8
+```
```

Comment:
here you might use some freebsd knowledge with 

```
gmake -j`echo "$(sysctl -n hw.ncpu) * 2" | bc`
```

Or at least mention why -j8 for those who aren't experienced. ;)

Reply:
@craigcomstock I don't know if that's necessary, `-j8` is very fast, on 1, 2, 4, 8, or 16 CPUs. (`-j4 and -j16` would be as well), it's not about exactly how many cores you have, just make it fast "in general". If you don't know what it means, you can just copy paste, or use the man page to find out.

Reply:
Could also do `gmake -j` to not set a specific limit.